spec/src/main/resources/META-INF/smithy/bsp/bsp.smithy

```diff
@@ -345,6 +346,18 @@ operation BuildTargetInverseSources {
     output: InverseSourcesResult
 }
 
+
+/// The wrapped sources request is sent from the client to the server to query for
+/// the list of build targets containing wrapped sources. Wrapped sources are script
+/// sources that are wrapped by the build tool with some top and bottom wrappers.
```

Comment:
To me as uninvolved person the use case is not really clear. Can you explain a bit more what this means and why it warrants a separate request?

Reply:
Basically scala scripts are not "real" scala code, they are text files that can become Scala code only after wrapping them with some code to make them real.

For example:
```scala
println("Hello world!")
```

doesn't compile in Scala.

But if you wrap the code with:
```scala
val top = s"""object main {
  def main(args: Array[String]): Unit = {
"""
```
and

```scala
val bottom = """  }
}
"""
```

it becomes valid Scala code.

Since it's not valid Scala code, IDEs can't parse them correctly, unless you tell them about this processing. scala-cli created this extension to have IDE supports on ammonite and scala-cli scripts.

Of course what I said should be better summarized and added to the documentation as javadcod and smithy comment.

> why it warrants a separate request?

This I don't know. I'm just following what's there since this is already used by scala-cli as a bespoke extension. If this can be achieved in a better way, let me know, I'm happy to update the PR.
I'm not very familiar with the BSP protocol other than some bugfixes I did in the Mill's implementation.